docs/src/main/tut/TypedML.md

```diff
@@ -15,17 +29,226 @@ spark.sparkContext.setLogLevel("WARN")
 
 import spark.implicits._
 ```
+
+## Example 1: predict a continuous value using a `TypedRandomForestRegressor`
+
+In this example, we want to predict `field3` of type Double from `field1` and `field2` using 
+a `TypedRandomForestRegressor`.
+
+### Training
+
+As with Spark ML API, we use a `TypedVectorAssembler` (the type-safe equivalent of `VectorAssembler`) to compute feature vectors:
+
+```tut:silent
+import frameless._
+import frameless.syntax._
+import frameless.ml._
+import frameless.ml.feature._
+import frameless.ml.regression._
+import org.apache.spark.ml.linalg.Vector
+```
+
+```tut:book
+case class Data(field1: Double, field2: Int, field3: Double)
+
+val trainingData = TypedDataset.create(
+  Seq.fill(10)(Data(0D, 10, 0D))
+)
+
+case class Features(field1: Double, field2: Int)
+val assembler = TypedVectorAssembler[Features]
+
+case class DataWithFeatures(field1: Double, field2: Int, field3: Double, features: Vector)
+val trainingDataWithFeatures = assembler.transform(trainingData).as[DataWithFeatures]
```

Comment:
As you had in the ticket, I think it's good to mention here that the `.as[DataWithFeatures]` is a type-safe casting.